documentation/lightgbm.md

```diff
@@ -0,0 +1,188 @@
+
+```
+search test {
+    rank-profile classify inherits default {
+        first-phase {
+            expression: lightgbm("lightgbm_model.json")
+        }
+    }
+}
+```
+
+Here, we specify that the model `lightgbm_model.json` (directly under the
+`models` directory) is applied to all documents matching a query which uses
+rank-profile `classify`. One important issue to consider is how to map features
+in the model to features that are available for Vespa to use in ranking.
+
+Take a look at the JSON file dumped from the example above:
+
+```
+{
+  "name": "tree",
+  "version": "v3",
+  "num_class": 1,
+  "num_tree_per_iteration": 1,
+  "label_index": 0,
+  "max_feature_idx": 1,
+  "average_output": false,
+  "objective": "binary sigmoid:1",
+  "feature_names": [
+    "feature_1",
+    "feature_2"
+  ],
+  "monotone_constraints": [],
+  "tree_info": [
+    ....
+  ],
+  "pandas_categorical": []
+}
+```
+
+Here, the section `feature_names` consists of the feature names used in the
+training set. When this model is evaluated in Vespa, Vespa expects that these
+feature names are valid [rank features](reference/rank-features.html).
+Examples are `attribute(field_name)` for a value that should be retrieved from
+a document, `query(name)` for a value that should be retrieved from the query,
+or possibly from other more complex rank features such as `fieldMatch(name)`.
+You can also define functions (which are valid rank features) with the LightGBM
+feature name to perform the mapping. An example:
+
+```
+search test {
+    document test {
+        field doc_attrib type double {
+            indexing: summary | attribute
+        }
+    }
+    rank-profile classify inherits default {
+        function feature_1() {
+            expression: attribute(doc_attrib)
+        }
+        function feature_2() {
+            expression: query(qurey_value)
+        }
+        first-phase {
+            expression: lightgbm("lightgbm_model.json")
+        }
+    }
+}
+
+```
+
+Here, when Vespa evaluates the model, it retrieves the value of `feature_1`
+from a document attribute called `doc_attrib`, and the value if `feature_2`
+from a query value passed along with the query.
+
+You could also use `attribute(doc_attrib)` directly as a feature name when
+training your LightGBM model. This allows you to dump rank features from Vespa
+to train a model directly. For more information, see [learning to
+rank](learning-to-rank.html).  Even though that page contains an example of
+using XGBoost, it is valid for LightGBM as well.
+
+If you have used XGBoost with Vespa previously, you might have noticed you
+have to wrap the `xgboost` feature in for instance a `sigmoid` function if you
+are using a binary classifier. That should not be needed in LightGBM as that
+information is passed along in the model dump as seen in the `objective` section
+in the JSON output above.
+
+## Using categorical features
```

Comment:
Could we have an example of the search definition feature map for a categorical feature?

Reply:
Good point. Added.